isis/src/control/objs/BundleAdjust/BundleAdjust.cpp

```diff
@@ -1896,234 +1880,68 @@ namespace Isis {
-      index++;
-    }
-
-    if (m_bundleSettings->solveTargetBody() && m_bundleTargetBody->solveTriaxialRadii()) {
-
-      std::vector<double> lookBWRTRadiusA =
-          measureCamera->GroundMap()->EllipsoidPartial(surfacePoint,
-                                                       CameraGroundMap::WRT_MajorAxis);
-
-      measureCamera->GroundMap()->GetdXYdPoint(lookBWRTRadiusA, &coeffTarget(0, index),
-                                               &coeffTarget(1, index));
-      index++;
-
-      std::vector<double> lookBWRTRadiusB =
-          measureCamera->GroundMap()->EllipsoidPartial(surfacePoint,
-                                                       CameraGroundMap::WRT_MinorAxis);
-
-      measureCamera->GroundMap()->GetdXYdPoint(lookBWRTRadiusB, &coeffTarget(0, index),
-                                               &coeffTarget(1, index));
-      index++;
-
-      std::vector<double> lookBWRTRadiusC =
-          measureCamera->GroundMap()->EllipsoidPartial(surfacePoint,
-                                                       CameraGroundMap::WRT_PolarAxis);
-
-      measureCamera->GroundMap()->GetdXYdPoint(lookBWRTRadiusC, &coeffTarget(0, index),
-                                               &coeffTarget(1, index));
-      index++;
-    }
-
-    index = 0;
-
-    if (observationSolveSettings->instrumentPositionSolveOption() !=
-        BundleObservationSolveSettings::NoPositionFactors) {
-
-      int numCamPositionCoefficients =
-          observationSolveSettings->numberCameraPositionCoefficientsSolved();
-
-      // Add the partial for the x coordinate of the position (differentiating
-      // point(x,y,z) - spacecraftPosition(x,y,z) in J2000
-      for (int cameraCoef = 0; cameraCoef < numCamPositionCoefficients; cameraCoef++) {
-        measureCamera->GroundMap()->GetdXYdPosition(SpicePosition::WRT_X, cameraCoef,
-                                                    &coeffImage(0, index),
-                                                    &coeffImage(1, index));
-        index++;
-      }
-
-      // Add the partial for the y coordinate of the position
-      for (int cameraCoef = 0; cameraCoef < numCamPositionCoefficients; cameraCoef++) {
-        measureCamera->GroundMap()->GetdXYdPosition(SpicePosition::WRT_Y, cameraCoef,
-                                                    &coeffImage(0, index),
-                                                    &coeffImage(1, index));
-        index++;
-      }
-
-      // Add the partial for the z coordinate of the position
-      for (int cameraCoef = 0; cameraCoef < numCamPositionCoefficients; cameraCoef++) {
-        measureCamera->GroundMap()->GetdXYdPosition(SpicePosition::WRT_Z, cameraCoef,
-                                                    &coeffImage(0, index),
-                                                    &coeffImage(1, index));
-        index++;
+    if (measureCamera->GetCameraType() != 6) {
```

Comment:
Why is this checking the camera type? It should just make the same calls for both CSM and ISIS, that's why we wanted to move the logic into the observation.

Reply:
Discussed in person -- the `CameraGroundMap()` doesn't exist for csm cameras, and we don't want to (a) do an image-to-ground call in each newly added partials function which would get called every time or (b) have a `observation::setupPartials()` function that gets called once for each observation, so we decided to leave this in. There's already a check to see if the camera is a framer, so it's not completely out of place to check if it's an Isis or Csm sensor.